spec/models/service_deployment_spec.rb

```diff
@@ -59,4 +59,37 @@
       end
     end
   end
+
+  describe '#generate_commit_link' do
+    context 'given a service' do
+      let(:user) do
+        User.create(id: 'abc123', name: 'test user', email: 'test@example.justice.gov.uk')
+      end
+
+      let(:service) do
+        Service.create!(id: 'fed456',
+                        name: 'My New Service',
+                        slug: 'my-new-service',
+                        git_repo_url: 'https://github.com/some-organisation/some-repository.git',
+                        created_by_user: user)
+      end
+
+      before do
+        subject.update_attributes(service: service, commit_sha: 'a123e56')
+      end
+
+      context 'when the git_repo_url exists' do
+        it 'sets the Github link for the commit sha' do
+          expect(subject.generate_github_link).to eq('https://github.com/some-organisation/some-repository/commit/a123e56')
+        end
+      end
+
+      context 'when the git_repo_url does not exist' do
+        it 'does not create a link for the commit sha' do
+          service.git_repo_url = ''
```

Comment:
This line should really be in a before block under the `context when the git_repo_url does not exist'` line - it's part of establishing the context rather than part of the test. That way, if any other tests are added later for 'when the git_repo_url does not exist', they'll all inherit the same context without needing to repeat the setup